ocaml-lsp-server/src/workspace_symbol.ml

```diff
@@ -0,0 +1,216 @@
+    =
+  let id = id_of_sig_item signature_item in
+  let loc = loc_of_sig_item signature_item in
+  let name = Ident.name id in
+  let kind = kind_of_sig_item signature_item in
+  let range = Range.of_loc (loc |> to_loc) in
+  let uri = sourcefile |> Uri.of_path in
+  let location = Location.create ~uri ~range in
+  SymbolInformation.create ~name ~kind ~location ()
+
+let find_build_dir path =
+  let ( / ) = Filename.concat in
+  let build_dir = "_build" in
+  let is_build folder = folder = build_dir in
+  let contains_build_dir = Array.exists ~f:is_build in
+  let home =
+    try Sys.getenv "HOME" with
+    | Not_found -> ""
+  in
+  let rec find path =
+    if path = home then
+      None
+    else
+      let files = Sys.readdir path in
+      if contains_build_dir files then
+        Some (path / build_dir)
+      else
+        let parent = Filename.dirname path in
+        if path = parent then
+          None
+        else
+          find parent
+  in
+  find path
+
+let subdirs dir =
+  let rec subdirs acc path =
+    Array.fold_left
+      ~f:(fun acc p ->
+        let path = Filename.concat path p in
+        if
+          try Sys.is_directory path with
+          | Sys_error _ -> false
+        then
+          subdirs acc path
+        else
+          acc)
+      ~init:(path :: acc) (Sys.readdir path)
+  in
+  subdirs [] dir
+
+let read_cmt root_uri cmt_file =
+  let cmt = Cmt_format.read_cmt cmt_file in
+  let sourcefile =
+    root_uri ^ "/" ^ Option.value ~default:"" cmt.cmt_sourcefile
+  in
+  let signatures = cmt_sign cmt in
+  signatures |> List.map ~f:(symbol_of_signature_item ~sourcefile)
+
+type cm_file =
+  | Cmt of string
+  | Cmti of string
+
+let string_of_cm cm =
+  match cm with
+  | Cmt f
+  | Cmti f ->
+    f
+
+module Cm_files = Map.Make (struct
+  type t = string
+
+  let compare a b = Stdlib.compare a b |> Ordering.of_int
+
+  let to_dyn t = Dyn.String t
+end)
+
+let find_dirfiles dirs =
+  let choose_file f1 f2 =
+    match (f1, f2) with
+    | (Cmt _ as f), _
+    | _, (Cmt _ as f) ->
+      f
+    | (Cmti _ as f), Cmti _ -> f
+  in
+  let split_filename file =
+    let maybe_index = String.rindex_opt file '.' in
+    let len = String.length file in
+    match maybe_index with
+    | None -> (file, "")
+    | Some index ->
+      let modul = String.capitalize_ascii (String.sub file ~pos:0 ~len:index) in
+      let ext =
+        String.lowercase_ascii
+          (String.sub file ~pos:(index + 1) ~len:(len - index - 1))
+      in
+      (modul, ext)
+  in
+  List.fold_left
+    ~f:(fun files dir ->
+      Sys.readdir dir
+      |> Array.fold_left
+           ~f:(fun acc file ->
+             let modul, ext = split_filename file in
+             let path = dir ^ "/" ^ file in
+             match ext with
+             | "cmt" -> Cm_files.set acc modul (Cmt path)
+             | "cmti" -> (
+               let current_file = Cm_files.find acc modul in
+               let cmti_file = Cmti path in
```

Comment:
Instead of looking cmti files, why don't we look in cmi files?

Reply:
I think I did that because of [this comment](https://github.com/OCamlPro/ocp-index/blob/master/libs/indexBuild.ml#L1263). My understanding is that it should be the same as `cmi` but with more info. To be honest, the difference between all of them is not clear yet to me 😅

Anyway, I guess `cmi` files should be fine too. I'll make the change :)